### PR TITLE
feat(auth): OTP via Email

### PR DESCRIPTION
**Pull request by Equartey**
*Description of changes:*
Adds OTP via Email 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/passwordless-auth-construct/src/services/ses_service.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      },
+      Source: this.config.fromAddress,
+    });
+
+    // Send Email via SES
+    const output = await this.sesClient.send(emailCommand);
+    logger.debug(`SMS sent: ${JSON.stringify(output, null, 2)}`);
+  };
+
+  public mask = (destination: string): string => {
```
Comment (Jordan-Nelson) [inline packages/passwordless-auth-construct/src/services/ses_service.ts:47]: *note[non-blocking]:* I am noticing that the logic here is not quite the same as Cognito. I don't think that is necessarily an issue but ideally they would be the same. I am going to create a task to investigate and align them if it makes sense.

**Review by Jordan-Nelson — APPROVED** [review]

File: packages/passwordless-auth-construct/package.json [hunk 2]
```diff
@@ -18,7 +18,8 @@
   },
   "dependencies": {
     "@aws-amplify/auth-construct-alpha": "^0.2.0-alpha.6",
-    "@aws-sdk/client-sns": "^3.429.0"
+    "@aws-sdk/client-sns": "^3.429.0",
+    "@aws-sdk/client-ses": "^3.445.0"
```
Comment (Jordan-Nelson) [inline packages/passwordless-auth-construct/package.json:22]: *Question:* Does`@aws-sdk/client-sns` need to be a higher version than `@aws-sdk/client-ses`? Should we align the two versions?

**Review by sobolk — APPROVED** [review]

File: packages/passwordless-auth-construct/package.json (hunk 2 above)
Comment (Equartey) [reply packages/passwordless-auth-construct/package.json:22]: They can be aligned. Will update 👍

File: packages/passwordless-auth-construct/src/otp/construct.ts [hunk 3]
```diff
@@ -30,6 +30,11 @@ export class AmplifyOtpAuth extends Construct {
         // see: https://docs.aws.amazon.com/sns/latest/dg/sns-using-identity-based-policies.html
         notResources: ['arn:aws:sns:*:*:*'],
       }),
+      // SES IAM policy
```
[the remark this answers is not in the document]
Comment (Jordan-Nelson) [reply packages/passwordless-auth-construct/src/otp/construct.ts:33]: I am going to mark this as resolved. We will need to refactor this a bit in the magic link PR.

**Pull request merged**